lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -982,6 +1081,30 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
         }
         return false;
       }
+
+      // Check the methods for consistency.
+      CXXRecordDecl::method_iterator Method2 = D2CXX->method_begin(),
+                                     Method2End = D2CXX->method_end();
+      for (CXXRecordDecl::method_iterator Method1 = D1CXX->method_begin(),
```

Comment:
Can we actually ensure that the order of the methods is the same?
Right now the order of importing messes up everything. After import, we artificially reorder the fields.
Do we want to do the same for methods? It might add more runtime overhead, so I am unsure. Maybe we are better off leaving the order of methods arbitrary and not check for structural equivalence at all.

Reply:
If order of methods is wrong the same problem would exist for fields and friends and maybe base classes too.
If the check is not done this can cause bad code imports.

Reply:
Yeah, but fields/friend are already reordered I think. We could always reorder more, but since right not it is a quadratic algorithm (due to linked lists being involved), we should keep the number of items to reorder minimal.

Reply:
`ASTNodeImporter::ImportDeclContext` does a reorder for every `Decl` that should include fields, friends, methods. Anyway, in structural equivalence check, it would be better to sort the items in some way.